### PR TITLE
Implement retweets

### PR DESCRIPTION
**Pull request by Saunved**

### CODE REVIEW

File: frontend/src/components/Tweet/SingleTweet.jsx [hunk 1]
```diff
@@ -5,30 +5,85 @@ import TweetActions from "@/components/Tweet/TweetActions";
...
+          setIsDeleted(true);
+        } else {
+          response.json().then((data) => {
+            console.error(data.message);
+          });
+        }
+        setIsLoading(true);
+      })
+      .catch((error) => {
+        console.error("Error liking tweet", error);
```
Comment (hritik5102) [inline frontend/src/components/Tweet/SingleTweet.jsx]: Error deleting tweets

File: frontend/src/components/Tweet/SingleTweet.jsx [hunk 2]
```diff
@@ -5,30 +5,85 @@ import TweetActions from "@/components/Tweet/TweetActions";
...
+import { Trash } from "phosphor-react";
+import { useEffect, useState } from "react";
+import session from "@/utils/session";
+import DeleteTweetModal from "@/components/DeleteTweetModal";
+import { BASE_URL } from "@/constants/routes";
 
 export default function Tweet({ tweet }) {
+  const [userHandle, setUserHandle] = useState(session.getUser().handle);
+  const [showModal, setShowModal] = useState(false);
+  const [isLoading, setIsLoading] = useState(false);
```
Comment (hritik5102) [inline frontend/src/components/Tweet/SingleTweet.jsx:17]: Where `isLoading` is used?

File: frontend/src/components/Tweet/TweetActions.jsx [hunk 3]
```diff
@@ -155,8 +164,8 @@ export default function TweetActions({ tweet }) {
         />
         <span className="text-sm">{likeCount || 0}</span>
       </button>
-      <ChartLine size={20} className="text-gray-700 hover:cursor-pointer" />
-      <Export size={20} className="text-gray-700 hover:cursor-pointer" />
+      {/* <ChartLine size={20} className="text-gray-700 hover:cursor-pointer" /> */}
+      {/* <Export size={20} className="text-gray-700 hover:cursor-pointer" /> */}
```
Comment (hritik5102) [inline frontend/src/components/Tweet/TweetActions.jsx]: Remove these two lines if not required

**Review by hritik5102 — APPROVED** [review]
Make following changes and then merge the changes

File: frontend/src/components/Tweet/SingleTweet.jsx (hunk 2 above)
Comment (Saunved) [reply frontend/src/components/Tweet/SingleTweet.jsx:17]: Added to button to disable it when the delete XHR is in progress

File: frontend/src/components/Tweet/SingleTweet.jsx (hunk 1 above)
Comment (Saunved) [reply frontend/src/components/Tweet/SingleTweet.jsx]: Done

**Pull request merged**